a10_octavia/controller/worker/tasks/a10_compute_tasks.py

```diff
@@ -39,16 +39,17 @@ def execute(self, amphora_id, loadbalancer=None,
                 server_group_id=None, ports=None, network_list=None):
 
         ports = ports or []
-        network_ids = CONF.a10_controller_worker.amp_boot_network_list[:]
-        if network_list:
-            for net in network_list:
-                if net not in network_ids:
-                    network_ids.append(net)
+        mgmt_net = CONF.a10_controller_worker.amp_mgmt_network
+        network_ids = {mgmt_net} if mgmt_net else {}
+        network_ids = network_ids.union(set(CONF.a10_controller_worker.amp_boot_network_list))
```

Comment:
**I think we need to make sure mgmt_net will still be the first one in the network_ids when pass to compute.build().**

Please confirm that:
When convert to set will the order be changed in the set. We need to make sure mgmt_net always be the first one in the set/list. otherwise vthunder may bootup using wrong mgmt network

Reply:
Good catch. The fact that sets are an unordered datatype slipped my mind when I was writing this code. I've added some logic to fix it and a unit test harness as well for each scenario.